PayItForward.Data.Services/StoriesService.cs

```diff
@@ -22,25 +22,78 @@ public StoriesService(IRepository<PayItForwardDbmodels.Story, Guid> storiesRepo,
             this.mapper = mapper;
         }
 
-        public IEnumerable<StoryDTO> GetStories(int take, int skip, string containsTitle = "")
+        public IEnumerable<StoryDTO> GetStories(int take, int skip, string containsTitle = "", string isFromCategory = "")
         {
-            var storiesFromDb = this.storiesRepo.GetAll()
+            var isCategoryNull = string.IsNullOrEmpty(isFromCategory);
+            List<PayItForwardDbmodels.Story> storiesFromDb;
+
+            if (!isCategoryNull)
+            {
+                storiesFromDb = this.storiesRepo.GetAll()
+                   .Include(user => user.User)
+                   .Include(category => category.Category)
+                   .Where(x => x.Title.Contains(containsTitle))
+                   .Where(st => st.Category.Name == isFromCategory && st.IsDeleted == false)
+                   .OrderBy(x => x.CreatedOn)
+                   .Skip(skip)
+                   .Take(take)
+                   .ToList();
+            }
+            else
+            {
+                storiesFromDb = this.storiesRepo.GetAll()
                .Include(user => user.User)
                .Include(category => category.Category)
-               .Where(x => x.Title.Contains(containsTitle))
+               .Where(x => x.Title.Contains(containsTitle) && x.IsDeleted == false)
                .OrderBy(x => x.CreatedOn)
                .Skip(skip)
                .Take(take)
                .ToList();
+            }
+
+            List<StoryDTO> stories = new List<StoryDTO>();
+            stories = this.mapper.Map<List<StoryDTO>>(storiesFromDb);
+            return stories;
+        }
+
+        public int CountStories(string containsTitle = "", string isFromCategory = "")
```

Comment:
Contains title can be changed to titleSubstring and isFromCategory to CategoryName or Category. Please review all the places where you have this, so to not put the same comment on all the places